kythe/rust/indexer/src/indexer/entries.rs

```diff
@@ -139,6 +139,9 @@ impl<'a> EntryEmitter<'a> {
         let mut diagnostic_vname = VName::new();
         let source_signature = source_vname.get_signature();
         diagnostic_vname.set_signature(format!("{}_{}", source_signature, sha256sum));
+        // Use the corpus and language from the source_vname
+        diagnostic_vname.set_corpus(source_vname.get_corpus().to_string());
+        diagnostic_vname.set_language(source_vname.get_language().to_string());
```

Comment:
It seems like this should be a copy of the source_vname, with a modified signature rather than a new vname with individually copied fields?